File: 10334.java
// Steven Kester Yuwono - UVa 10334

import java.util.Scanner;
import java.math.BigInteger;

class Main {
	static BigInteger fib[] = new BigInteger[1001];
	public static void init() {
		fib[0] = BigInteger.ONE;
		fib[1] = BigInteger.valueOf(2);
		for(int i=2;i<=1000;i++){
			fib[i] = fib[i-2].add(fib[i-1]);
		}
	}

	public static void main (String[] args) {
		init();
		Scanner cin = new Scanner(System.in);
		while(cin.hasNext()) {
			int n = cin.nextInt();
			System.out.println(fib[n]);
		}
	}
}

File: 10334.cpp
// Steven Kester Yuwono - UVa 10334

#include <cstdio>
using namespace std;

unsigned long long fib[1001];

void init() {
	fib[0] = 1;
	fib[1] = 2;
	for(int i=2;i<=1000;i++){
		fib[i] = fib[i-1] + fib[i-2];
	}
}

int main() {
	init();
	int n;
	while (scanf("%d",&n)==1) {
		printf("%lld\n",fib[n]);
	}
	return 0;
}
